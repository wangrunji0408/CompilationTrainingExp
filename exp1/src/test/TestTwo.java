package test;

public class TestTwo {
    public static int frobble(int x) {
	return 8-x;
    }

    public static void main (String [] args) {
	int a=0, b=0, c=0, d;

	for (a=0; a<7; a++) {

	    for (d=0; d<7; d++) {
	        c=frobble(c);
	    }
				
	    if (a-b == 2) {
		c++;
	    } else if (c == 4) {
		break;
	    }
	}

	c++;

    }
}
