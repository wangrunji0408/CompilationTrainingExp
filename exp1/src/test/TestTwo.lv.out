Now analyzing test.TestTwo
Method: frobble
Initialization completed.
entry: [R0]
1 in:  [R0]
1 out: [T1]
2 in:  [T1]
2 out: []
exit: []
Method: main
Initialization completed.
entry: []
1 in:  []
1 out: []
2 in:  []
2 out: []
3 in:  []
3 out: [R3]
4 in:  [R3]
4 out: [R1, R3]
5 in:  []
5 out: []
6 in:  []
6 out: []
7 in:  []
7 out: []
8 in:  []
8 out: []
9 in:  []
9 out: []
10 in:  []
10 out: []
11 in:  []
11 out: []
12 in:  []
12 out: []
13 in:  []
13 out: []
14 in:  []
14 out: []
15 in:  []
15 out: []
16 in:  []
16 out: []
17 in:  []
17 out: []
18 in:  []
18 out: []
19 in:  []
19 out: []
20 in:  []
20 out: []
21 in:  []
21 out: []
22 in:  []
22 out: []
23 in:  [R1, R3]
23 out: [R1, R3]
24 in:  []
24 out: []
25 in:  [R1, R3]
25 out: [R1, R3, R4]
26 in:  [R3]
26 out: []
27 in:  []
27 out: []
28 in:  []
28 out: []
29 in:  []
29 out: []
30 in:  []
30 out: []
31 in:  []
31 out: []
32 in:  []
32 out: []
33 in:  []
33 out: []
34 in:  [R1, R3, R4]
34 out: [R1, R3, R4]
35 in:  []
35 out: []
36 in:  []
36 out: []
37 in:  []
37 out: []
38 in:  [R1, R3, R4]
38 out: [R1, R4, T5]
39 in:  [R1, R4, T5]
39 out: [R1, R3, R4]
40 in:  [R1, R3, R4]
40 out: [R1, R3, R4]
41 in:  [R1, R3, R4]
41 out: [R1, R3, R4]
42 in:  [R1, R3]
42 out: [R1, R3, T5]
43 in:  [R1, R3, T5]
43 out: [R1, R3]
44 in:  []
44 out: []
45 in:  []
45 out: []
46 in:  []
46 out: []
47 in:  [R1, R3]
47 out: [R1, R3]
48 in:  [R1, R3]
48 out: [R1, R3]
49 in:  [R1, R3]
49 out: [R1, R3]
50 in:  [R1, R3]
50 out: [R1, R3]
51 in:  [R1, R3]
51 out: [R1, R3]
52 in:  [R3]
52 out: [R3]
exit: []
Method: <init>
Initialization completed.
entry: [R0]
1 in:  [R0]
1 out: []
2 in:  [R0]
2 out: [R0]
3 in:  []
3 out: []
exit: []
